File: backend/services/script-service/app/schemas/case.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

# 案例基础模型
class CaseBase(BaseModel):
    title: str
    description: str
    author: str
    tags: List[str]
    coverColor: str

# 案例创建请求
class CaseCreateRequest(CaseBase):
    pass

# 案例更新请求
class CaseUpdateRequest(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    author: Optional[str] = None
    tags: Optional[List[str]] = None
    coverColor: Optional[str] = None

# 案例响应
class CaseResponse(CaseBase):
    id: str
    likes: int
    views: int
    createdAt: str
    updatedAt: str

    class Config:
        from_attributes = True

# 案例列表响应
class CaseListResponse(BaseModel):
    cases: List[CaseResponse]
    total: int
    page: int
    page_size: int

# 点赞响应
class LikeResponse(BaseModel):
    message: str
    likes: int

# 浏览响应
class ViewResponse(BaseModel):
    message: str
    views: int

# 分享响应
class ShareResponse(BaseModel):
    message: str

File: frontend/src/types/case.ts
// 案例广场 (Case Square) 类型定义

/** 单个案例 */
export interface CaseItem {
  id: string
  title: string
  description: string
  author: string
  likes: number
  views: number
  tags: string[]
  coverColor: string
  videoUrl?: string
  createdAt: string
  updatedAt: string
}

/** 案例列表响应 */
export interface CaseListResponse {
  cases: CaseItem[]
  total: number
  page: number
  pages: number
}

/** 案例列表请求参数 */
export interface CaseListParams {
  page?: number
  pageSize?: number
  tag?: string
  sortBy?: 'views' | 'likes' | 'createdAt'
  order?: 'asc' | 'desc'
}
